src/kakarot/instructions/system_operations.cairo

```diff
@@ -428,10 +437,14 @@ namespace CallHelper {
 
         // Load calldata from Memory
         let (calldata: felt*) = alloc();
-        let memory = Memory.load_n(
+        let (memory, gas_cost) = Memory.expand_and_load_n(
             self=ctx.memory, element_len=args_size, element=calldata, offset=args_offset
         );
 
+        // TODO: account for value_to_empty_account_cost in dynamic gas
+        let dynamic_gas = gas_cost + SystemOperations.GAS_COST_COLD_ADDRESS_ACCESS + SystemOperations.GAS_COST_POSITIVE_VALUE;
```

Comment:
Shouldn’t we use `value` here to check >0?

Reply:
Sure